src/core/HazelcastError.ts

```diff
@@ -22,45 +22,101 @@ export interface HazelcastErrorConstructor {
 }
 
 /**
- * Base class for all specific errors thrown by Hazelcast client.
+ * Represents a stack trace element of server-side exception.
+ */
+export interface ServerErrorStackElement {
+
+    /**
+     * Class name.
+     */
+    className: string;
+
+    /**
+     * Method name.
+     */
+    methodName: string;
+
+    /**
+     * Name of the file containing the class.
+     */
+    fileName: string;
+
+    /**
+     * Line number from the class.
+     */
+    lineNumber: number;
+
+}
+
+/**
+ * Represents server-side exception.
+ */
+export interface ServerError {
```

Comment:
I think there are some duplication and unnecessary information here. 

`errorCode` does not mean a lot to the users. It is already used to instantiate the error.
`message` is the same as the error's message.
`className`  is almost the same as the error's class name.

I believe the only interesting part here is the `stackTraceElements`. Maybe, we could rename `HazelcastError`s `serverError` property with something like `serverStackTrace`. What do you think about it?

Reply:
Fair point. I've removed `ServerError` interface and left only the `serverStackTrace: ServerErrorStackElement[]` property `HazelcastError`. Done in 08d69ff0e778006bc18f01d447d78a4c6464238c